addons/purchase/tests/test_purchase.py

```diff
@@ -942,3 +942,54 @@ def test_vendor_price_by_purchase_order_company(self):
         self.assertEqual(po.amount_untaxed, 15.0)
         po.company_id = company_a.id
         self.assertEqual(po.amount_untaxed, 10.0)
+
+    def test_bill_in_purchase_matching_individual(self):
+        """
+        Tests that if the vendor is an individual with a company, the bill will still appear when
+        using the purchase matching button on a vendor bill
+        """
+        company_partner = self.env['res.partner'].create({
+            'name': 'Small Company',
+            'company_type': 'company',
+        })
+        individual_partner = self.env['res.partner'].create({
+            'name': 'Big Vendor',
+            'company_type': 'person',
+            'parent_id': company_partner.id,
+        })
+        downpayment_product = self.env['product.product'].create({
+            'name': 'Downpayment Service',
+            'type': 'service',
+            'invoice_policy': 'order',
+        })
+        purchase_order = self.env['purchase.order'].create({
+            'partner_id': individual_partner.id,
+            'company_id': self.env.company.id,
+            'order_line': [(0, 0, {
+                'product_id': downpayment_product.id,
+                'name': 'Downpayment Service',
+                'product_qty': 1,
+                'price_unit': 100,
+                'date_planned': '2025-01-01',
+            })]
+        })
```

Comment:
```suggestion
        purchase_order = self.env['purchase.order'].create({
            'partner_id': individual_partner.id,
            'order_line': [(0, 0, {
                'product_id': downpayment_product.id,
                'product_qty': 1,
                'price_unit': 100,
            })]
        })
```
IMO It makes it more readable when avoiding noisy values :)